src/helpers/react-docs.ts

```diff
@@ -0,0 +1,121 @@
+import fetch from "node-fetch";
+import { gitHubToken } from "./env";
+
+const LOOKUP_REGEX = /<Intro>\s*(.*?)\s*<\/Intro>/gs;
+const LINK_REGEX = /\[([^\]]+)\]\((?!https?:\/\/)([^)]+)\)/g;
+
+const BASE_URL =
+  "https://api.github.com/repos/reactjs/react.dev/contents/src/content/reference/";
+
+export const getReactDocsContent = async (searchPath: string) => {
+  try {
+    const response = await fetch(`${BASE_URL}${searchPath}.md`, {
+      method: "GET",
+      headers: {
+        Accept: "application/vnd.github+json",
+        Authorization: `Bearer ${gitHubToken}`,
+        "X-GitHub-Api-Version": "2022-11-28",
+      },
+    });
+    const json = await response.json();
+    const contentBase64 = json.content;
+    const decodedContent = Buffer.from(contentBase64, "base64").toString(
+      "utf8",
+    );
+    return processReactDocumentation(decodedContent);
+  } catch (error) {
+    console.error("Error:", error);
+    return null;
+  }
+};
+
+export const getReactDocsSearchKey = (search: string) => {
+  const normalizedSearch = search.toLowerCase();
+
+  return REACT_AVAILABLE_DOCS.find((key) => {
+    const keyParts = key.split("/");
+    const name = keyParts[keyParts.length - 1];
+    const namespace =
+      keyParts.length <= 2 ? key : `${keyParts[0]}/${keyParts[2]}`;
+
+    return (
+      namespace.toLowerCase() === normalizedSearch ||
+      name.toLowerCase() === normalizedSearch
+    );
+  });
+};
+
+const processReactDocumentation = (content: string) => {
+  const patchedContentLinks = content.replace(LINK_REGEX, (_, text, link) => {
+    return `[${text}](https://react.dev${link})`;
+  });
+
+  const matches = [...patchedContentLinks.matchAll(LOOKUP_REGEX)];
+
+  if (matches.length > 0) {
+    const [introContent] = matches.map(([, match]) => match.trim());
+    return introContent;
+  }
+
+  return null;
+};
+
+const REACT_AVAILABLE_DOCS = [
```

Comment:
I am not quite sure about having constant defined here and not schedule task top update it, but the react documentation updates to file path are not frequent and this may become out of date only after major releases.

Do we want to have additional requests to populate this array or having an generate script would be enough, when necessary? What do you think?

Reply:
Yeah, I think this is a reasonable assumption. If we find it breaks more frequently than is convenient then we can reevaluate, but given effort to repair breakage + likely frequency of breakage, this seems reasonable